src/interfaces/IAuthorizedEvent.ts

```diff
@@ -55,7 +63,10 @@ export function isAuthorizedEventBase<P> (
 ): event is IAuthorizedEventBase {
   return (
     event != null &&
-    typeof event.httpMethod === 'string' &&
+    (typeof event.httpMethod === 'string' ||
+      (typeof event.requestContext === 'object' &&
+        event.requestContext !== null &&
+        typeof event.requestContext.http === 'object')) &&
```

Comment:
Please extract type guards with names based on the interface names you choose above to make this check more readable.

Reply:
I've updated this and split them into `isAuthorizedRestApiGatewayEvent` and `isAuthorizedHttpApiGatewayEvent` :+1: